include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -45,73 +45,73 @@ def ForeignType : FIRRTLDialectType<CPred<"!$_self.isa<FIRRTLType>()">,
                                     "foreign type", "::mlir::Type">;
 
 def ClockType :
-  FIRRTLDialectType<CPred<"$_self.isa<ClockType>()">, "clock",
-                    "::circt::firrtl::ClockType">;
+  FIRRTLDialectType<CPred<"firrtl::type_isa<ClockType>($_self)">, "clock",
+                    "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::ClockType>">;
 
 def NonConstClockType :
-  FIRRTLDialectType<CPred<"$_self.isa<ClockType>()">, "clock",
+  FIRRTLDialectType<CPred<"firrtl::type_isa<ClockType>($_self)">, "clock",
                     "::circt::firrtl::ClockType">,
   BuildableType<"::circt::firrtl::ClockType::get($_builder.getContext())">;
 
-def IntType : FIRRTLDialectType<CPred<"$_self.isa<IntType>()">,
- "sint or uint type", "::circt::firrtl::IntType">;
+def IntType : FIRRTLDialectType<CPred<"firrtl::type_isa<IntType>($_self)">,
+ "sint or uint type", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::IntType>">;
 
-def SIntType : FIRRTLDialectType<CPred<"$_self.isa<SIntType>()">,
- "sint type", "::circt::firrtl::SIntType">;
+def SIntType : FIRRTLDialectType<CPred<"firrtl::type_isa<SIntType>($_self)">,
+ "sint type", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::SIntType>">;
 
-def UIntType : FIRRTLDialectType<CPred<"$_self.isa<UIntType>()">,
- "uint type", "::circt::firrtl::UIntType">;
+def UIntType : FIRRTLDialectType<CPred<"firrtl::type_isa<UIntType>($_self)">,
+ "uint type", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::UIntType>">;
 
-def AnalogType : FIRRTLDialectType<CPred<"$_self.isa<AnalogType>()">,
- "analog type", "::circt::firrtl::AnalogType">;
+def AnalogType : FIRRTLDialectType<CPred<"firrtl::type_isa<AnalogType>($_self)">,
+ "analog type", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::AnalogType>">;
 
-def BundleType : FIRRTLDialectType<CPred<"$_self.isa<BundleType>()">,
- "BundleType", "::circt::firrtl::BundleType">;
-def OpenBundleType : FIRRTLDialectType<CPred<"$_self.isa<OpenBundleType>()">,
+def BundleType : FIRRTLDialectType<CPred<"firrtl::type_isa<BundleType>($_self)">,
+ "BundleType", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::BundleType>">;
+def OpenBundleType : FIRRTLDialectType<CPred<"firrtl::type_isa<OpenBundleType>($_self)">,
  "OpenBundleType", "::circt::firrtl::OpenBundleType">;
 
-def FVectorType : FIRRTLDialectType<CPred<"$_self.isa<FVectorType>()">,
-  "FVectorType", "::circt::firrtl::FVectorType">;
-def OpenVectorType : FIRRTLDialectType<CPred<"$_self.isa<OpenVectorType>()">,
+def FVectorType : FIRRTLDialectType<CPred<"firrtl::type_isa<FVectorType>($_self)">,
+  "FVectorType", "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::FVectorType>">;
+def OpenVectorType : FIRRTLDialectType<CPred<"firrtl::type_isa<OpenVectorType>($_self)">,
```

Comment:
Probably can't have a type alias to an open aggregate (at least in practice, since they contains probes..), but checking this is fine however that should mean the C++ class type for this should be `BaseTypeAliasOr` which it can't be, so probably drop this.